backend/src/test/scala/com/softwaremill/adopttapir/starter/StarterServiceITTest.scala

```diff
@@ -3,49 +3,206 @@ package com.softwaremill.adopttapir.starter
+object Setup {
+  val validConfigurations = for {
     effect <- EffectRequest.values
     server <- ServerImplementationRequest.values
     docs <- List(true, false)
     metrics <- List(true, false)
     json <- JsonImplementationRequest.values
     starterRequest = StarterRequest("myproject", "com.softwaremill", effect, server, addDocumentation = docs, addMetrics = metrics, json)
     starterDetails <- FormValidator.validate(starterRequest).toSeq
-  } {
-    it should s"return zip file containing working sbt folder with: $effect/$server/docs=$docs/metrics=$metrics/$json" in {
-      val service = createStarterService
-      sbtCompileTest(service.generateZipFile(starterDetails))
+  } yield starterDetails
+
+  implicit class StarterDetailsWithDescribe(details: StarterDetails) {
+    lazy val describe: String =
+      s"${details.serverEffect}/${details.serverImplementation}/docs=${details.addDocumentation}/metrics=${details.addMetrics}/${details.jsonImplementation}"
+  }
+
+  type TestFunction = Integer => Unit
+}
+
+class StarterServiceITTest extends BaseTest with ParallelTestExecution with BeforeAndAfterAll {
+  import Setup._
+
+  for { details <- Setup.validConfigurations } {
+    it should s"return zip file containing working sbt folder with: ${details.describe}" in {
+      val backend: SttpBackend[Identity, Any] = HttpURLConnectionBackend()
+
+      // define endpoints integration tests
+      val helloEndpointTest: Option[TestFunction] = Some(port => {
+        val result: String =
+          basicRequest.response(asStringAlways).get(uri"http://localhost:${port}/hello?name=Frodo").send(backend).body
+
+        result shouldBe "Hello Frodo"
+      })
+
+      val docsEndpointTest: Option[TestFunction] = testOption(
+        details.addDocumentation,
+        port => {
+          val result: String =
+            basicRequest.response(asStringAlways).get(uri"http://localhost:${port}/docs/docs.yaml").send(backend).body
+
+          result should include("paths:\n  /hello:")
+        }
+      )
+
+      val metricsEndpointTest: Option[TestFunction] = testOption(
+        details.addMetrics,
+        port => {
+          val result: String =
+            basicRequest.response(asStringAlways).get(uri"http://localhost:${port}/metrics").send(backend).body
+
+          result should (include("# HELP tapir_request_duration_seconds Duration of HTTP requests")
+            and include("# TYPE tapir_request_duration_seconds histogram")
+            and include("# HELP tapir_request_total Total HTTP requests")
+            and include("# TYPE tapir_request_total counter")
+            and include("# HELP tapir_request_active Active HTTP requests")
+            and include("# TYPE tapir_request_active gauge"))
+        }
+      )
+
+      // get implementation for a given configuration, compile it, unit & integration test it
+      val service = ServiceUnderTest(details)
+      service.run(helloEndpointTest, docsEndpointTest, metricsEndpointTest)
```

Comment:
Service run should take only assertion functions which exists. 

This reduce number of things which Service should do.